### ISSUE TITLE
after the end of the unrest in the colony, the colonists are not in their places

### ISSUE THREAD

**Issue by MrZorG33** [We-the-People-civ4col-mod/Mod#538]
after the unrest ends in the colony, the colonists do not return to the places of work where they were before the start of the unrest.
for example: a lumberjack and a free colonist are not in the places where they were before the unrest.
![после беспорядков колонисты не на своих местах](https://user-images.githubusercontent.com/61798883/151249082-6661c636-0872-4cb4-8905-4b17baad1c86.jpg)

as can be seen from the colonist in the town hall and the carpenter, their places of work have been fixed (lock button).

**Comment by ErikSandbraaten** [comment]
Yeah, this is a shortcoming of the job assigner. It does not consider the colonist's LBD (learning by doing) status

**Comment by ErikSandbraaten** [comment]
Wait a sec, were the colonists "locked" before the unrest ?

**Comment by MrZorG33** [comment]
yes, that's the problem.

**Comment by ErikSandbraaten** [comment]
I think the culprit is that when a locked colonist is removed from a plot, there is no memory concerning the plot he worked so he/she will be reassigned by the jobassigner.

**Comment by ErikSandbraaten** [comment]
Implementing a fix now. (though it is more like a new feature that preserves the job of locked citizens when bumped )

**Comment by Nightinggale** [comment]
This bug has been reported for 4.0.1 meaning either the fix didn't work or it was never committed into any branch used for the release.

**Comment by raystuttgart** [comment]
I do not think that it was commited to "New Hope" at that time.

**Comment by MrZorG33** [comment]
@devolution79, could you test this fix?

**Comment by MrZorG33** [comment]
the disposition of the colonists before the unrest
![image](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/3d3b3bbe-9f50-481c-97d6-986624c4285c)
[before the end of the unrest.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/12928037/before.the.end.of.the.unrest.zip)

the disposition of the colonists after the end of the unrest
![image](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/9ca46813-dfcd-4442-bc09-04cb226cef3b)
[after the end of the unrest.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/12928051/after.the.end.of.the.unrest.zip)

**Comment by vfauth** [comment]
This is still an issue, is the above fix working?
